### PR TITLE
Add Trino 434 release notes

### LINKED ISSUE #19704: Release notes for 434

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for the 434 release.

## Additional context and related issues

Fixes #19704 

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 10 Nov 2023

* #19692 ✅ rn ✅ docs
* #19703 ✅ rn ✅ docs
* #19680 ✅ rn ✅ docs
* #19691 ✅ rn ✅ docs
* #19676 ✅ rn ✅ docs

## 11 Nov 2023

* #19513 ✅ rn ✅ docs
* #19664 ✅ rn ✅ docs
* #19709 ✅ rn ✅ docs

## 12 Nov 2023

* #19713 ✅ rn ✅ docs

## 13 Nov 2023

* #19714 ✅ rn ✅ docs
* #19683 ✅ rn ✅ docs
* #19655 ✅ rn ✅ docs
* #19718 ✅ rn ✅ docs
* #19650 ✅ rn ✅ docs
* #19711 ✅ rn ✅ docs

## 14 Nov 2023

* #19667 ✅ rn ✅ docs
* #19667 ✅ rn ✅ docs
* #19728 ✅ rn ✅ docs
* #19734 ✅ rn ✅ docs
* #19736 ✅ rn ✅ docs
* #19726 ✅ rn ✅ docs
* #19719 ✅ rn ✅ docs
* #19742 ✅ rn ✅ docs
* #19732 ✅ rn ✅ docs
* #19723 ✅ rn ✅ docs
* #19746 ✅ rn ✅ docs
* #19739 ✅ rn ✅ docs

## 15 Nov 2023

* #19735 ❌ @findepi rn ? ✅ docs
* #19750 ✅ rn ✅ docs
* #19743 ✅ rn ✅ docs
* #19745 ✅ rn ✅ docs
* #19756 ✅ rn ✅ docs
* #19720 ✅ rn ✅ docs
* #19712 ✅ rn ✅ docs
* #19752 ✅ rn ✅ docs
* #19766 ✅ rn ✅ docs
* #19769 ✅ rn ✅ docs

## 16 Nov 2023

* #19588 ✅ rn ✅ docs
* #19776 ✅ rn ✅ docs
* #19699 ✅ rn ✅ docs
* #19757 ✅ rn ✅ docs
* #19774 ✅ rn ✅ docs
* #19773 ✅ rn ✅ docs
* #19767 ✅ rn ✅ docs
* #19772 ✅ rn ✅ docs
* #19782 ✅ rn ✅ docs
* #19786 ✅ rn ✅ docs

## 17 Nov 2023

* #19777 ✅ rn ✅ docs
* #19763 ✅ rn ✅ docs
* #19727 ✅ rn ✅ docs
* #19791 ✅ rn ✅ docs
* #19790 ✅ rn ✅ docs
* #19780 ✅ rn ✅ docs
* #19797 ✅ rn ✅ docs
* #19804 ✅ rn ✅ docs
* #19800 ✅ rn ✅ docs

## 18 Nov 2023

* #19787 ✅ rn ✅ docs
* #19807 ✅ rn ✅ docs
* #19809 ✅ rn ✅ docs
* #19812 ✅ rn ✅ docs
* #19577 ✅ rn ✅ docs

## 19 Nov 2023

* #19808 ✅ rn ✅ docs

## 20 Nov 2023

* #19803 ✅ rn ✅ docs
* #19801 ✅ rn ✅ docs
* #19801 ✅ rn ✅ docs
* #19824 ✅ rn ✅ docs
* #19793 ✅ rn ✅ docs
* #19823 ✅ rn ✅ docs
* #19759 ✅ rn ✅ docs
* #19693 ✅ rn ✅ docs
* #19831 ✅ rn ✅ docs
* #19710 ✅ rn ✅ docs
* #19768 ✅ rn ✅ docs
* #19802 ✅ rn ✅ docs
* #18507 ✅ rn ✅ docs

## 21 Nov 2023

* #19830 ✅ rn ✅ docs
* #19838 ✅ rn ✅ docs
* #19796 ✅ rn ✅ docs
* #19846 ✅ rn ✅ docs
* #19850 ✅ rn ✅ docs
* #19842 ✅ rn ✅ docs
* #19841 ✅ rn ✅ docs
* #19852 ✅ rn ✅ docs
* #19851 ✅ rn ✅ docs

## 22 Nov 2023

* #18897 ✅ rn ✅ docs
* #19845 ✅ rn ✅ docs
* #19849 ✅ rn ✅ docs
* #19794 ✅ rn ✅ docs
* #19765 ✅ rn ✅ docs
* #19762 ✅ rn ✅ docs
* #19858 ✅ rn ✅ docs
* #19856 ✅ rn ✅ docs
* #19868 ✅ rn ✅ docs

## 23 Nov 2023

* #19867 ✅ rn ✅ docs
* #19855 ✅ rn ✅ docs
* #19862 ✅ rn ✅ docs
* #19859 ✅ rn ✅ docs in #19881
* #19870 ✅ rn ✅ docs
* #19872 ✅ rn ✅ docs
* #19853 ✅ rn ✅ docs
* #19853 ✅ rn ✅ docs
* #19761 ✅ rn ✅ docs
* #19779 ✅ rn ✅ docs
* #19871 ✅ rn ✅ docs
* #19866 ✅ rn ✅ docs

## 24 Nov 2023

* #19878 ✅ rn ✅ docs
* #19863 ✅ rn ✅ docs
* #19891 ✅ rn ✅ docs
* #19881 ✅ rn ✅ docs
* #19894 ✅ rn ✅ docs

## 25 Nov 2023

* #19899 ✅ rn ✅ docs

## 26 Nov 2023

* #19875 ✅ rn ✅ docs
* #19792 ✅ rn ✅ docs
* #19901 ✅ rn ✅ docs

## 27 Nov 2023

* #19874 ✅ rn ✅ docs
* #19877 ✅ rn ✅ docs
* #19848 ✅ rn ✅ docs
* #19896 ✅ rn ✅ docs
* #19883 ✅ rn ✅ docs
* #19911 ✅ rn ✅ docs
* #19922 ✅ rn ✅ docs
* #19893 ✅ rn ✅ docs
* #19921 ✅ rn ✅ docs
* #19857 ✅ rn ✅ docs
* #19927 ✅ rn ✅ docs
* #19925 ✅ rn ✅ docs
* #19928 ✅ rn ✅ docs
* #19900 ✅ rn ✅ docs
* #19789 ✅ rn ✅ docs
* #19814 ✅ rn ✅ docs

## 28 Nov 2023

* #19929 ✅ rn ✅ docs
* #19905 ✅ rn ✅ docs
* #19930 ✅ rn ✅ docs
* #19933 ✅ rn ✅ docs
* #18984 ✅ rn ✅ docs
* #19890 ✅ rn ✅ docs
* #19879 ✅ rn ✅ docs
* #19861 ✅ rn ✅ docs
* #19916 ✅ rn ✅ docs
* #19681 ✅ rn ✅ docs
* #19889 ✅ rn ✅ docs

## 29 Nov 2023

* #19948 ✅ rn ✅ docs
* #19951 ✅ rn ✅ docs
* #19953 ✅ rn ✅ docs
* #19961 ✅ rn ✅ docs
* #19880 ✅ rn ✅ docs
* #19960 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-434.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Release 434 (17 Nov 2023)
+
+## General
+
+* {{breaking}} Rename the `query.max-writer-tasks-count` configuration property
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-434.md:5]: fyi @radek-starburst

File: docs/src/main/sphinx/release/release-434.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+## SingleStore connector
+
+## SQL Server connector
+
+## TPC-H connector
+
+## SPI
+
+* Add bulk append methods to `BlockBuilder`. ({issue}`19577`)
+* {{breaking}} Remove `VariableWidthBlockBuilder` `buildEntry` method. ({issue}`19577`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-434.md:102]: @dain 

or should we use 

```
VariableWidthBlockBuilder#buildEntry
```

File: docs/src/main/sphinx/release/release-434.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+## Hive connector
+
+* Add timestamp to date schema evolution. ({issue}`19513`)
+* Add tracing to metastore access. ({issue}`19712`)
+
+## Hudi connector
+
+## Iceberg connector
+
+* {{breaking}} Remove support for legacy table statistics tracking. ({issue}`19803`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-434.md:55]: fyi @findepi

File: docs/src/main/sphinx/release/release-434.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# Release 434 (20 Nov 2023)
+
+## General
+
+* {{breaking}} Rename the `query.max-writer-tasks-count` configuration property
+  and the related session property `max_writer_tasks_count` to
```
Comment (colebow) [inline docs/src/main/sphinx/release/release-434.md:6]:
```suggestion
  and the related `max_writer_tasks_count` session property to
```

File: docs/src/main/sphinx/release/release-434.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+## CLI
+
+## Accumulo connector
+
+## Atop connector
+
+## BigQuery connector
+
+* Support reading `json` type. ({issue}`19790`)
```
Comment (colebow) [inline docs/src/main/sphinx/release/release-434.md:27]: Maybe "Add support for reading `json` columns" instead?

File: docs/src/main/sphinx/release/release-434.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+## Blackhole connector
+
+## Cassandra connector
+
+## ClickHouse connector
+
+## Delta Lake connector
+
+* Improve performance when reading large checkpoint files on partitioned tables.
```
Comment (colebow) [inline docs/src/main/sphinx/release/release-434.md:37]: Should probably include a link to documentation on what checkpoint files are.

File: docs/src/main/sphinx/release/release-434.md (hunk 2 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-434.md:102]: I'd normally do `VariableWidthBlockBuilder#buildEntry`

File: docs/src/main/sphinx/release/release-434.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+## Druid connector
+
+## Elasticsearch connector
+
+## Google Sheets connector
+
+## Hive connector
+
+* Add timestamp to date schema evolution. ({issue}`19513`)
```
Comment (colebow) [inline docs/src/main/sphinx/release/release-434.md:48]: I think I've been documenting this as support for reading columns that have been changed from x type to y type, so we might want to change past notes or this note so they're consistent. Also, `timestamp` and `date`.

**Review by colebow — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-434.md (hunk 6 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:37]: not sure where to link to - seems like we used https://docs.delta.io/latest/delta-batch.html#-data-retention .. but thats kinda odd

Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:37]: and we didnt link in some other release notes either ..

File: docs/src/main/sphinx/release/release-434.md (hunk 7 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:48]: Done

File: docs/src/main/sphinx/release/release-434.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:102]: seems like we used dot in the past..

File: docs/src/main/sphinx/release/release-434.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# Release 434 (22 Nov 2023)
+
+## General
+
+* {{breaking}} Rename the `query.max-writer-tasks-count` configuration property
+  and the related `max_writer_tasks_count` session property to
+  `query.max-writer-task-count` and `max_writer_task_count`. ({issue}`19793`)
+* Prevent table function failure when its invocation is a first query run on a
+  worker. ({issue}`19849`)
+* Prevent sudden increases in PagePartitioner memory consumption in some
```
Comment (martint) [inline docs/src/main/sphinx/release/release-434.md:10]: PagePartitioner is an implementation detail. What's the user-visible condition that causes this? (e.g., specific query shapes, features, etc)

File: docs/src/main/sphinx/release/release-434.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+## Delta Lake connector
+
+* Improve performance when reading large checkpoint files on partitioned tables.
+  ({issue}`19588`)
+
+## Hive connector
+
+* Add support for columns that changed from `timestamp` to `date` type.
+  ({issue}`19513`)
+* Add tracing to metastore access. ({issue}`19712`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-434.md:27]: This is an internal detail. I don't think we need to document it here.

File: docs/src/main/sphinx/release/release-434.md (hunk 8 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md]: PR is large .. but I reworded to mention affected queries.. we might have to hedge with "and similar scenarios" or so

File: docs/src/main/sphinx/release/release-434.md (hunk 9 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md]: Removed

File: docs/src/main/sphinx/release/release-434.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+## BigQuery connector
+
+* Add support for reading `json` columns. ({issue}`19790`)
+* Add support for `DELETE` statement. ({issue}`6870`)
+* Improve performance when writing rows. ({issue}`18897`)
+
+## Delta Lake connector
+
+* Improve performance when reading large checkpoint files on partitioned tables. ({issue}`19588`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-434.md:24]: @ebyhr @findepi @findinpath  .. you suggested the same text for two different but related PRs .. do we want to just link to one PR? Or have one  entry with two issue/Pr links? Or reword one entry? cc @martint @colebow

Comment (findepi) [reply docs/src/main/sphinx/release/release-434.md:29]: > do we want to just link to one PR?

link to both please

File: docs/src/main/sphinx/release/release-434.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+  and the related `max_writer_tasks_count` session property to
+  `query.max-writer-task-count` and `max_writer_task_count`. ({issue}`19793`)
+* Improve performance of `INSERT ... SELECT` queries that contain a redundant
+  `ORDER BY` clause. ({issue}`19916`)
+* Fix incorrect results for queries involving comparisons between `double` and
+  `real` zero and negative zero. ({issue}`19828`)
+* Fix performance regression caused by suboptimal scalar subqueries planning. ({issue}`19922`)
+* Prevent table function failure when its invocation is a first query run on a
+  worker. ({issue}`19849`)
+* Prevent sudden increases in memory consumption in some queries using
```
Comment (martint) [inline docs/src/main/sphinx/release/release-434.md:16]: Typo in "JOINT".

Also, there's no such a thing as CROSS JOIN UNNEST. Cross join and unnest are two separate constructs. Therefore, I would rephrase it as "join involving UNNEST".

File: docs/src/main/sphinx/release/release-434.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+* Add support for separate metadata caching configuration for schemas, tables,
+  and metadata. ({issue}`19859`)
+
+## Delta Lake connector
+
+* {{breaking}} Stop accepting invalid, previously ignored Hive catalog
+  configuration parameters. ({issue}`19735`)
+* Improve performance when reading large checkpoint files on partitioned tables.
+  ({issue}`19588`, {issue}`19848`)
+* Pushdown filters involving columns of type `TIMESTAMP WITH TIME ZONE`. ({issue}`18664`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-434.md:36]: "Push down" (verb)

File: docs/src/main/sphinx/release/release-434.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+* Fix query failure when reading Parquet column index for timestamp columns. ({issue}`16801`)
+
+## Hudi connector
+
+* Fix query failure when reading Parquet column index for timestamp columns. ({issue}`16801`)
+
+## Iceberg connector
+
+* {{breaking}} Remove support for legacy table statistics tracking. ({issue}`19803`)
+* {{breaking}} Stop accepting invalid, previously ignored Hive catalog
```
Comment (martint) [inline docs/src/main/sphinx/release/release-434.md:56]: Same as above

File: docs/src/main/sphinx/release/release-434.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+* Add support for a `FILTER` clause to the `LISTAGG` function. ({issue}`19869`)
+* {{breaking}} Rename the `query.max-writer-tasks-count` configuration property
+  and the related `max_writer_tasks_count` session property to
+  `query.max-writer-task-count` and `max_writer_task_count`. ({issue}`19793`)
+* Improve performance of `INSERT ... SELECT` queries that contain a redundant
+  `ORDER BY` clause. ({issue}`19916`)
+* Fix incorrect results for queries involving comparisons between `double` and
+  `real` zero and negative zero. ({issue}`19828`)
+* Fix performance regression caused by suboptimal scalar subqueries planning. ({issue}`19922`)
+* Prevent table function failure when its invocation is a first query run on a
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:14]: Updated

File: docs/src/main/sphinx/release/release-434.md [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+* Improve performance when writing rows. ({issue}`18897`)
+
+## ClickHouse connector
+
+* Add support for separate metadata caching configuration for schemas, tables,
+  and metadata. ({issue}`19859`)
+
+## Delta Lake connector
+
+* {{breaking}} Stop accepting invalid, previously ignored Hive catalog
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:32]: These Hive configs were previously allowed in Delta Lake and Iceberg .. and now they are not anymore .. so it breaks for those two connectors

[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:32]: And the fact that they are Hive configs details that subset ..

[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:32]: But for now I left that out and people can read the PR I guess.

File: docs/src/main/sphinx/release/release-434.md (hunk 12 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md:36]: Done

File: docs/src/main/sphinx/release/release-434.md (hunk 11 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-434.md]: Updated
